frontend/ui/widget/doublespinwidget.lua

```diff
@@ -221,11 +229,16 @@ function DoubleSpinWidget:update()
         },
         {
             text = self.ok_text,
+            enabled = self.left_value ~= left_widget:getValue() or self.right_value ~= right_widget:getValue(),
             callback = function()
                 if self.callback then
-                    self.callback(left_widget:getValue(), right_widget:getValue())
+                    self.left_value = left_widget:getValue()
+                    self.right_value = right_widget:getValue()
```

Comment:
No need to save the values / update self.left/right_value when there is no `self.callback` ?

Reply:
> when there is no self.callback

I just wanted to ask, what is the reason of this check? How can be a spinwidget without callback on ok_button?

Reply:
Dunno, I don't have all the answers :)
May be it is just a safety check.
May be it was based from some other widget that has an optional callback.
It could even be used as a read only display stuff (like Book status) (that I know we don't have that use any spin widget).

(Usually, I don't question what's there, I just follow the existing logic, try to fit what I add in what's already there :)  I rarely feel the need to clean up, rationalize - with the risk involved of breaking things :)

Your current code would mean that self.left_value and right_value has no use when there is no self.callback - which is fine if that's really the case.

Reply:
Let's update self values on pressing ok_button, with or without self.callback.